src/main/webapp/maps_geolocation.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * {@code addGoToMyLocationControl} allows the user to center the map
+ * at their location by clicking a button.
+ * {@code addEnableGeolocationControl} allows the user to turn on/off
+ * the geolocation feature, which displays the user's current location continuously.
+ */
+
+var currentLocationMarker = null;
+var trackUser = false;
+var watchPositionId;
+var currentLocation;
+
+const NO_GEOLOCATION_SUPPORT_MSG = "The browser doesn't support geolocation.";
+const ENABLE_GEOLOCATION_MSG = "Please enable geolocation first!";
+const NO_LOCATION_PERMISSION_MSG = "Please allow location permission!.";
+const NO_LOCATION_INFORMATION_MSG = "Location information is unavailable.";
+const REQUEST_TIMEOUT_MSG = "The request to get user location timed out.";
+const UNKNOWN_ERROR_MSG = "An unknown error occurred while geolocating.";
+
+const GEOLOCATION_IMG_ID = "enableGeolocationIcon";
+const ENABLE_GEOLOCATION_TITLE = "Enable geolocation";
+const DISABLED_GEOLOCATION_IMG_SRC = "./img/geolocation.svg"
+const DISABLE_GEOLOCATION_TITLE = "Disable geolocation";
+const ENABLE_GEOLOCATION_IMG_SRC = "./img/geolocation_active.svg";
+
+const GO_TO_MY_LOCATION_IMG_ID = "goToMyLocationIcon";
+const GO_TO_MY_LOCATION_TITLE = "Go to my location";
+
+/**
+ * Adds a button to the map which turns on/off geolocation.
+ * Geolocation can be battery-consuming, and the user should be able
+ * to tund it off.
+ * Remark that the audio-guide creation process doesn't require the user's location at all.
+ */
+function addEnableGeolocationControl(map) {
+  const geolocationControlDiv =
+      createControlDiv(ENABLE_GEOLOCATION_TITLE, DISABLED_GEOLOCATION_IMG_SRC, GEOLOCATION_IMG_ID);
+  geolocationControlDiv.index = 1;
+  map.controls[google.maps.ControlPosition.RIGHT_BOTTOM].push(geolocationControlDiv);
+  trackUser = false;
+  geolocationControlDiv.addEventListener("click",
+          event => geolocationControlEvent(map, geolocationControlDiv));
+}
+
+/**
+ * When the geolocation button gets clicked, its status is toggled.
+ * If geolocation is on, then each time the user's loctaion changes,
+ * and event will be triggered,
+ * and the currentLocationMarker and currentlocation will be reset
+ */
+function geolocationControlEvent(map, geolocationControlDiv) {
```

Comment:
Usually methods that react to events begin their name with "on". So here it would be "onGeoLocationControlEvent".

Reply:
Changed.